### ISSUE TITLE
Mongodb error 11000

### ISSUE THREAD

**Issue by Houtek** [Graylog2/graylog2-server#7080]
## Expected Behavior
No error in the log file

## Current Behavior
Every hour at 1 minute past full hour I get the error in the logs:

```
ERROR [TrafficCounterCalculator] Uncaught exception in periodical
com.mongodb.DuplicateKeyException: Write failed with error code 11000 and error message 'E11000 duplicate key error collection: graylog.traffic index: bucket_1 dup key: { : new Date(1578283200000) }'
```
The full error message is here: https://pastebin.com/pyJgQUTP

## Steps to Reproduce (for bugs)
It appeared first on 2019-07-18, then disappeared and reappeared on 2019-12-07. I saw it today the first time.

## Your Environment

* Graylog Version: 3.1.3
* Elasticsearch Version: 6.8.6
* MongoDB Version: 3.6.16
* Operating System:  CentOS Linux release 7.7.1908 (Core)

**Comment by bernd** [comment]
@Houtek Can you please show me the output of the following commands executed on the MongoDB server:

- `mongo graylog --eval "db.traffic.find().pretty()"`
- `mongo graylog --eval "db.traffic.getIndexes()"`

(Please replace `graylog` with the actual database name if it isn't named graylog in your setup)

That should help us to debug the issue. Thank you

Refs #6339

**Comment by Houtek** [comment]
@bernd 

Thank your for looking into this.

```
rs0:PRIMARY> db.traffic.getIndexes()
[
        {
                "v" : 2,
                "key" : {
                        "_id" : 1
                },
                "name" : "_id_",
                "ns" : "graylog.traffic"
        },
        {
                "v" : 2,
                "unique" : true,
                "key" : {
                        "bucket" : 1
                },
                "name" : "bucket_1",
                "ns" : "graylog.traffic"
        }
]
rs0:PRIMARY> 
```
[mongodb-output_1.txt](https://github.com/Graylog2/graylog2-server/files/4030189/mongodb-output_1.txt)
Please find the output of the last 3471 lines attached.

**Comment by bernd** [comment]
@Houtek We believe we found the root cause of this. It has been documented in https://github.com/Graylog2/graylog2-server/issues/7258.

A fix for that has been committed and will be included in the upcoming 3.2.0 release.

Thank you for the report!
